examples/MQ/pixelDetector/macros/run_dAsciiSource.C

```diff
@@ -26,19 +47,17 @@ void run_dAsciiSource(TString mcEngine = "TGeant3")
     // -----   Timer   --------------------------------------------------------
     TStopwatch timer;
 
-    PixelEventHeader* pixelEventHeader = new PixelEventHeader();
-
     // -----   Reconstruction run   -------------------------------------------
-    FairRunAna* fRun = new FairRunAna();
-    fRun->SetEventHeader(pixelEventHeader);
-    fRun->SetSink(new FairRootFileSink(outFile));
+    FairRunAna run{};
+    run.SetEventHeader(std::make_unique<PixelEventHeader>());
+    run.SetSink(new FairRootFileSink(outFile));
```

Comment:
I think we should indicate the ownership transferring in our example code:
```suggestion
auto outFile = std::make_unique<FairRootFileSink>();
run.SetSink(outFile.release());
```

Reply:
We already have a `SetSink(unique_ptr)`. So you're right, I should use that!

We don't have that for `SetSource` yet, because SetSource is a `virtual` thing which we need to understand first.

Reply:
I have updated the commit to use unique_ptr for FairSink. But not yet for FairSource.

Let's please postpone the SetSource thing until we understand that mess a bit better.

Reply:
But fSource in the FairRun is already a unqiue_ptr:

https://github.com/FairRootGroup/FairRoot/blob/fa7108b45684c729c2e88b12d7fbf4bbb5a4dc65/base/steer/FairRun.h#L243-L245

Why can we not have a setter taking a unique_ptr if the FairSource is an abstract class?

Reply:
Oh, sorry, I understood incorrectly. I didn't see SetSource is a virtual function. 

This is bad. Setters and getters should not be a virtual function.

Reply:
The original SetSink is virtual too, but luckily noone (in FairRoot) overrides it. We probably should remove the virtual. But as we want to deprecate it anyway…

Reply:
I think the `FairRun::SetSource` has been made `virtual` only to implement `FairRunAnaProof::SetSource`.
I vote for removing `virtual`.